<launch>
    <arg name="fcu_conn" default="uart"/>
    <arg name="fcu_ip" default="127.0.0.1"/>
    <arg name="gcs_bridge" default="tcp"/>
    <arg name="viz" default="true"/>
    <arg name="web_server" default="true"/>
    <arg name="web_video_server" default="true"/>
    <arg name="rosbridge" default="true"/>
    <arg name="bottom_camera" default="false"/>
    <arg name="fpv_camera" default="true"/>
    <arg name="fpv_camera_device" default="/dev/v4l/by-id/usb-HD_Camera_Manufacturer_USB_2.0_Camera-video-index0"/>
    <arg name="arduino" default="false"/>

    <!-- mavros -->
    <include file="$(find clever)/launch/mavros.launch">
        <arg name="fcu_conn" value="$(arg fcu_conn)"/>
        <arg name="fcu_ip" value="$(arg fcu_ip)"/>
        <arg name="gcs_bridge" value="$(arg gcs_bridge)"/>
        <arg name="viz" value="$(arg viz)"/>
    </include>

    <!-- web server -->
    <include file="$(find clever)/launch/web_server.launch" if="$(arg web_server)"/>

    <!-- web video server -->
    <node name="web_video_server" pkg="web_video_server" type="web_video_server" if="$(arg web_video_server)"/>

    <!-- TODO: marker position estimator -->

    <!-- simplified offboard control -->
    <node name="simple_offboard" pkg="clever" type="simple_offboard.py"/>

    <!-- bottom camera -->
    <include file="$(find clever)/launch/bottom_camera.launch" if="$(arg bottom_camera)"/>

    <!-- rosbridge -->
    <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" if="$(arg rosbridge)"/>

    <!-- FPV video streaming -->
    <include file="$(find clever)/launch/fpv_camera.launch" if="$(arg fpv_camera)">
        <arg name="device" value="$(arg fpv_camera_device)"/>
    </include>

    <!-- Arduino bridge -->
    <include file="$(find clever)/launch/arduino.launch" if="$(arg arduino)"/>
</launch>
